### PR TITLE
replace sync pause with async pause for nxsig_process

### PR DESCRIPTION
**Pull request by hujun260**
## Summary
In the kernel, we are planning to remove all occurrences of up_cpu_pause as one of the steps to simplify the implementation of critical sections. The goal is to enable spin_lock_irqsave to encapsulate critical sections, thereby facilitating the replacement of critical sections(big lock) with smaller spin_lock_irqsave(small lock)
## Impact
up_schedule_sigaction
## Testing
Configuring NuttX and compile:
$ ./tools/configure.sh -l qemu-armv8a:nsh_smp
$ make
Running with qemu
$ qemu-system-aarch64 -cpu cortex-a53 -smp 4 -nographic -machine virt,virtualization=on,gic-version=3
-net none -chardev stdio,id=con,mux=on -serial chardev:con -mon chardev=con,mode=readline -kernel ./nuttx

### CODE REVIEW

**Comment by xiaoxiang781216** [comment]
@hujun260 please rebase your patch to fix the conflict

**Review by xiaoxiang781216 — APPROVED** [review]

File: include/nuttx/sched.h [hunk 1]
```diff
@@ -716,6 +716,11 @@ struct tcb_s
 
   struct xcptcontext xcp;                /* Interrupt register save area    */
 
+  /* The following function pointer is non-zero if there are pending signals
+   * to be processed.
+   */
+
+  CODE void *sigdeliver;                 /* Actual type is sig_deliver_t */
```
Comment (xiaoxiang781216) [inline include/nuttx/sched.h:723]: let's change to:
```
 sig_deliver_t  sigdeliver;
```
and remove all `(sig_deliver_t)`

File: sched/signal/sig_dispatch.c [hunk 2]
```diff
@@ -118,7 +165,37 @@ static int nxsig_queue_action(FAR struct tcb_s *stcb, siginfo_t *info)
           if (!stcb->sigdeliver)
             {
               stcb->sigdeliver = nxsig_deliver;
```
Comment (xiaoxiang781216) [inline sched/signal/sig_dispatch.c:167]: need move after the variable declaration at line 169 and 170

File: sched/signal/sig_dispatch.c [hunk 3]
```diff
@@ -118,7 +165,37 @@ static int nxsig_queue_action(FAR struct tcb_s *stcb, siginfo_t *info)
...
+                      arg.saved_affinity = stcb->affinity;
+                      arg.need_restore   = true;
+
+                      stcb->flags        |= TCB_FLAG_CPU_LOCKED;
+                      CPU_SET(stcb->cpu, &stcb->affinity);
+                    }
+
+                  arg.pid = stcb->pid;
+                  nxsched_smp_call_single(stcb->cpu, sig_handler, &arg,
+                                          true);
```
Comment (xiaoxiang781216) [inline sched/signal/sig_dispatch.c:192]: merge to previous line

File: arch/arm/src/arm/arm_schedulesigaction.c [hunk 4]
```diff
@@ -75,70 +75,63 @@
...
+   */
 
-  if (!tcb->sigdeliver)
-    {
-      tcb->sigdeliver = sigdeliver;
-
-      /* First, handle some special cases when the signal is
-       * being delivered to the currently executing task.
-       */
+  sinfo("rtcb=%p current_regs=%p\n", this_task(),
```
Comment (xiaoxiang781216) [inline arch/arm/src/arm/arm_schedulesigaction.c:86]: merge log with line 80, please fix ALL similar places.

File: arch/arm/src/arm/arm_schedulesigaction.c [hunk 5]
```diff
@@ -75,70 +75,63 @@
...
+       * restored by the signal trampoline after the signal has been
+       * delivered.
+       */
 
-          tcb->xcp.regs            = (void *)
-                                     ((uint32_t)tcb->xcp.regs -
-                                                XCPTCONTEXT_SIZE);
-          memcpy(tcb->xcp.regs, tcb->xcp.saved_regs, XCPTCONTEXT_SIZE);
+      tcb->xcp.regs            = (void *)
+                                  ((uint32_t)tcb->xcp.regs -
```
Comment (xiaoxiang781216) [inline arch/arm/src/arm/arm_schedulesigaction.c:120]: align

File: sched/signal/sig_dispatch.c (hunk 3 above)
Comment (hujun260) [reply sched/signal/sig_dispatch.c:192]: will report "error: Long line found"

**Review by xiaoxiang781216 — APPROVED** [review]

**Comment by masayuki2009** [comment]
@hujun260 
I noticed that sighand_test always hangs with qemu-intel64:nsh (non-SMP) after merging this PR.


```
user_main: signal handler test
sighand_test: Initializing semaphore to 0
sighand_test: Unmasking SIGCHLD
sighand_test: Registering SIGCHLD handler
sighand_test: Starting waiter task
sighand_test: Started waiter_main pid=59
waiter_main: Waiter started
waiter_main: Unmasking signal 32
waiter_main: Registering signal handler
waiter_main: oact.sigaction=0 oact.sa_flags=0 oact.sa_mask=0000000000000000
waiter_main: Waiting on semaphore
sighand_test: Signaling pid=59 with signo=32 sigvalue=42

```

**Comment by hujun260** [comment]
> @hujun260 I noticed that sighand_test always hangs with qemu-intel64:nsh (non-SMP) after merging this PR.
> 
> ```
> user_main: signal handler test
> sighand_test: Initializing semaphore to 0
> sighand_test: Unmasking SIGCHLD
> sighand_test: Registering SIGCHLD handler
> sighand_test: Starting waiter task
> sighand_test: Started waiter_main pid=59
> waiter_main: Waiter started
> waiter_main: Unmasking signal 32
> waiter_main: Registering signal handler
> waiter_main: oact.sigaction=0 oact.sa_flags=0 oact.sa_mask=0000000000000000
> waiter_main: Waiting on semaphore
> sighand_test: Signaling pid=59 with signo=32 sigvalue=42
> ```

ok i'll look into this issue

**Comment by hujun260** [comment]
https://github.com/apache/nuttx/pull/13888
fix this issue

**Pull request merged**